custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -168,14 +168,18 @@ def get_excel_data(self):
             }
             if not self.national_user:
                 user_location = logged_in_user_locations[loop_counter]
+                user_location_type_name = \
+                    self.get_location_id_string_from_location_type(user_location.location_type_name)
                 location_type_filter[self.get_location_id_string_from_location_type(
                     user_location.location_type_name)]: user_location.get_id
+            else:
+                user_location_type_name = None
 
             all_awc_locations = AwcLocation.objects.filter(**location_type_filter).values(*self.required_fields)
             # converting the result set to matrix to fetch ancestors for a given location
             location_matrix, location_ids =\
-                self.convert_rs_to_matrix(all_awc_locations, self.get_location_id_string_from_location_type(
-                    user_location.location_type_name))
+                self.convert_rs_to_matrix(all_awc_locations, user_location_type_name)
```

Comment:
What happens if `user_location_type_name` is None in this function?

Reply:
If it is not None , it will fetch locations under that given location else will fetch all the locations assuming it is a national user

Reply:
Does it filter out test locations?

Reply:
Is there any filter param for test locations. As of now i did not add any

Reply:
@calellowitz  I think filtering in the dashboard users in next steps will eliminate the test location. Because its very unlikely we follow the same username pattern for test users. But certainly possible, So agree upon explicitly excluding test locations just to be extra safe.

@challabeehyv  you can filter out the location by checking `<location>_is_test` is that location will be a test location then this column value will be `1`.

Reply:
@Rohit25negi 
I am not sure i understand your comment. For national users with access to all locations (which this fix is for), they will see the test locations. What filter are you talking about

Reply:
Test locations are not supposed to be seen by anyone on the dashboard. @challabeehyv

Reply:
added